src/main/java/frc/robot/Constants.java

```diff
@@ -114,7 +107,17 @@ public static void initSwerve() {
                 () -> swerveDrive.getEstimator().getEstimatedPosition(),
                 swerveDrive::resetPose,
                 swerveDrive::getCurrentSpeeds,
-                (speeds) -> swerveDrive.drive(speeds, false),
+                (speeds) -> {
+                    // Fixes diversion from note during autonomous
+                    if (Vision.getInstance().getYawToNote().isPresent()
+                            && RobotContainer.getInstance().isIntaking) {
+                        speeds.vyMetersPerSecond =
+                                SwerveConstants.vyOffsetControllerAutonomous.calculate(
+                                        Vision.getInstance().getYawToNote().getAsDouble(), 0);
+                    }
+
+                    swerveDrive.drive(speeds, false);
```

Comment:
Maybe extract this into an external method so we could use it in teleop if we'd like